### PR TITLE
Add new ValidationErr enum | update tests

### PR DESCRIPTION
**Pull request by HuanDay**
#19

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -8,6 +8,8 @@ extern crate alloc;
 extern crate std as alloc;
 
 use alloc::borrow::ToOwned;
+use alloc::fmt;
```
Comment (holygits) [inline src/lib.rs]: `use alloc::{self, Display, Formatter};`

File: src/lib.rs [hunk 2]
```diff
@@ -17,6 +19,48 @@ use pact::interpreter::{interpret, types::PactType};
...
+            Domain::Method => write!(f, "method"),
+            Domain::MethodArguments => write!(f, "method arguments"),
+            Domain::Module => write!(f, "module"),
+        }
+    }
+}
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum ValidationErr {
+    NoPermission(Domain),
```
Comment (holygits) [inline src/lib.rs]: Nice!

File: src/lib.rs [hunk 3]
```diff
@@ -17,6 +19,48 @@ use pact::interpreter::{interpret, types::PactType};
 
 mod test;
 
+#[derive(Debug, Eq, PartialEq)]
+pub enum Domain {
```
Comment (holygits) [inline src/lib.rs]: Add a doc comment

File: src/lib.rs [hunk 4]
```diff
@@ -17,6 +19,48 @@ use pact::interpreter::{interpret, types::PactType};
...
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            Domain::Method => write!(f, "method"),
+            Domain::MethodArguments => write!(f, "method arguments"),
+            Domain::Module => write!(f, "module"),
+        }
+    }
+}
+
+#[derive(Debug, Eq, PartialEq)]
```
Comment (holygits) [inline src/lib.rs]: Add a doc comment

File: src/lib.rs [hunk 5]
```diff
@@ -17,6 +19,48 @@ use pact::interpreter::{interpret, types::PactType};
...
+            ValidationErr::ConstraintsInterpretation => {
+                write!(f, "error while interpreting constraints")
+            }
+        }
+    }
+}
+
+pub trait Validate {
+    fn validate(&self, module: &str, method: &str, args: &[PactType]) -> Result<(), ValidationErr>;
+}
```
Comment (elim0322) [inline src/lib.rs]: Thanks for moving this! 👍

File: src/lib.rs (hunk 1 above)
Comment (holygits) [reply src/lib.rs]: Sorry, missed the `fmt` part: `use alloc::fmt::{self, Display, Formatter};`

**Pull request merged**